src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -0,0 +1,431 @@
+                                                       GetApexRuntimeRoot() + "/etc/tz/", // Android 10+, Fallback location if the above isn't found or corrupted
+                                                       Environment.GetEnvironmentVariable("ANDROID_DATA") + "/misc/zoneinfo/",
+                                                       Environment.GetEnvironmentVariable("ANDROID_ROOT") + DefaultTimeZoneDirectory};
+                foreach (var tzFileDir in tzFileDirList)
+                {
+                    string tzFilePath = Path.Combine(tzFileDir, TimeZoneFileName);
+                    if (LoadData(tzFilePath))
+                    {
+                        _tzFileDir = tzFileDir;
+                        _tzFilePath = tzFilePath;
+                        return;
+                    }
+                }
+
+                throw new TimeZoneNotFoundException(SR.TimeZoneNotFound_ValidTimeZoneFileMissing);
+            }
+
+            private bool LoadData(string path)
+            {
+                if (!File.Exists(path))
+                {
+                    return false;
+                }
+                try {
+                    using (FileStream fs = File.OpenRead(path))
+                    {
+                        LoadTzFile(fs);
+                    }
+                    return true;
+                }
+                catch (IOException) {}
+
+                return false;
+            }
+
+            private unsafe void LoadTzFile(Stream fs)
+            {
+                int headerSize = 24;
+                Span<byte> buffer = stackalloc byte[headerSize];
+                int bytesRead = 0;
+                int bytesLeft = headerSize;
+
+                while (bytesLeft > 0)
+                {
+                    int b = fs.Read(buffer);
+                    if (b == 0)
+                    {
+                        break;
+                    }
+
+                    bytesRead += b;
+                    bytesLeft -= b;
+                }
+
+                AndroidTzDataHeader header = LoadHeader(buffer);
+                ReadIndex(fs, header.indexOffset, header.dataOffset);
+            }
+
+            private unsafe AndroidTzDataHeader LoadHeader(Span<byte> buffer)
+            {
+                // tzdata files are expected to start with the form of "tzdata2012f\0" depending on the year of the tzdata used which is 2012 in this example
+                // since we're not differntiating on year, check for tzdata and the ending \0
+                Span<byte> signature = buffer.Slice(0, 12);
+                Span<byte> tzDataSignature = stackalloc byte[8];
+                tzDataSignature.Clear();
+
+                signature.Slice(0, 6).CopyTo(tzDataSignature);
+                ulong tzdata = (ulong)TZif_ToInt64(tzDataSignature);
+
+                char[] cbits = Encoding.Default.GetChars(tzDataSignature.ToArray());
```

Comment:
`cbits` doesn't look to be used.

I don't understand why we are copying anything out of the buffer here at all for the signature. Just check the buffer directly for the data you expect, throw if it isn't  expected, and move on. There is no reason to copy this data around just to compare it.

Reply:
Thanks, removed cbits and sigBits, they were used for tracing/debugging

Right, is the problem that `CopyTo` was used? It felt that comparing byte by byte was not concise, and we wanted to have something similar to `TZifHead`. Would the below block be fine, or is there also some performance hit when slicing the byte Span instead of just directly comparing as `if (buffer[0] != (byte)'t' || buffer[1] != (byte)'z' || buffer[2] != (byte)'d' || buffer[3] != (byte)'a' || buffer[4] != (byte)'t' || buffer[5] != (byte)'a' || buffer[11] != 0)`
```
Span<byte> tz = (ushort)TZif_ToInt16(buffer.Slice(0, 2));
Span<byte> data = (uint)TZif_ToInt32(buffer.Slice(2, 4));

if (tz != 0x747A || data != 0x64617461 || signature[11] != 0)
{
    // 0x747A  0x646174640000 = {0x74, 0x7A} {0x64, 0x61, 0x74, 0x64} = "tz" "data"
```

Reply:
>  Would the below block be fine, or is there also some performance hit when slicing the byte Span instead of just directly comparing

I don't see a performance problem with reading a ushort and a uint from the buffer and comparing the values. That approach would be perfectly fine IMO.

(note the code doesn't compile since you are casting to ushort and assigning to a `Span<byte>` variable, but the compiler would have told you that 😄 )